compaction/compaction_manager.cc

```diff
@@ -1024,16 +1024,22 @@ future<> compaction_manager::perform_sstable_scrub(column_family* cf, sstables::
 }
 
 future<> compaction_manager::remove(column_family* cf) {
+    auto handle = _compaction_state.extract(cf);
```

Comment:
I see a lot of code use `get_compaction_state()` which can just reinsert it. Do you intend to only prevent a specific code part to get it from here on?

Reply:
Unfortunately, the current design created implicitly using `_compaction_state[cf]`, and I didn't want to change that from the grounds up.

This hunk intention is to prevent `can_proceed` from finding the compaction state.  `can_proceed` calls `try_get_compaction_state` that returns if a disengaged `compaction_state_ptr` if not found and doesn't implicitly create a new one.

Reply:
sounds like a good idea for a follow up.
It should be easy also after the next version (preview https://github.com/bhalevy/scylla/commits/harden-compaction_manager-remove-v2)
that I'll submit once it passes all tests

Reply:
Done in v2